### PR TITLE
Deployment automation

### PR DESCRIPTION
**Pull request by johnpcooke94**
This PR adds a new /docs/ directory to the root of the project for containing non-sensitive documentation. Currently it only has one document in it, `deployment.md`. This document details our server hardware and software architecture, and the details of how to manually deploy the application. Additionally, it lays out details for how automated deployments are handled via self-hosted GitHub Actions runners.

This PR also includes a GitHub Actions workflow for automatically deploying new versions of the backend to the server. NOTE: This currently only works to deploy new versions of the software on a machine that is already hosting it - on a fresh machine, the first deployment will need to be done manually.

I've also committed the Kestrel service file for the backend to this repository for easy access later, in case it should be lost elsewhere.

### CODE REVIEW

**Review by Cliftonz — DISMISSED** [review]
Lgtm

File: cslabs-backend.service [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Description=CS Labs Backend Kestrel Service
+
+[Service]
+WorkingDirectory=/var/www/cslabs-backend/CSLabs.Api/bin/Release/net6.0/publish
+ExecStart=/usr/bin/dotnet /var/www/cslabs-backend/CSLabs.Api/bin/Release/net6.0/publish/CSLabs.Api.dll
+Restart=always
+# Restart service after 10 seconds if the dotnet service crashes:
+RestartSec=3
+KillSignal=SIGINT
+SyslogIdentifier=dotnet-example
```
Comment (jasekiw) [inline cslabs-backend.service]: Can the syslog identifier be set to our application name so we can find the logs easier?

File: doc/deployment.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Currently, the application uses Nginx as its webserver. The application is hosted on internal virtual
+machines managed via Proxmox on the internal cluster. There is one virtual machine for the dev environment
+and another for production.
+
+## Nginx
+
+Nginx must be installed on the server and configured to direct requests to the appropriate location. The
+configuration file that configured Nginx to do this correctly is committed to the back-end repository, in
+the root directory. `nginx.conf.dev` is the configuration for the dev environment, and `nginx.conf` is the
```
Comment (jasekiw) [inline doc/deployment.md:22]: Can you name this `nginx.dev.conf`? That will help the editor know it's a conf file.

**Review by jasekiw — CHANGES_REQUESTED** [review]

File: doc/deployment.md (hunk 2 above)
Comment (johnpcooke94) [reply doc/deployment.md:22]: You know what I think I forgot to commit this file anyway 😅

I'll get it committed under `nginx.dev.conf` later tonight.

File: cslabs-backend.service (hunk 1 above)
Comment (johnpcooke94) [reply cslabs-backend.service:12]: Is just `cslabs` good or should it be `cslabs-backend`?

Comment (johnpcooke94) [reply cslabs-backend.service]: Done, pushed a commit.

**Review by jasekiw — APPROVED** [review]

**Pull request merged**